### PR TITLE
Fixes Floating Messages

### PR DESCRIPTION
**Pull request by nevimer**
## About The Pull Request

IT WAS A # ALL ALONG

(no its not actually tilted ingame)
dreamseeker_xXIN96muRJ

## Why It's Good For The Game

Makes things WORK AS INTENDED

## Changelog

:cl:
fix: floating messages now have their personality
/:cl:

### CODE REVIEW

File: code/game/atoms/atoms.dm [hunk 1]
```diff
@@ -824,18 +824,18 @@
...
-		italics = TRUE
 	for(var/mob/M in get_mobs_in_view(message_range, src))
 		if(M.client)
 			speech_bubble_hearers += M.client
 	if(length(speech_bubble_hearers))
-		INVOKE_ASYNC(src, /atom/movable/proc/animate_chat, message, speaking, italics, speech_bubble_hearers, 30)
+		send_runechat_to_atom_movable(message = message, speaking = speaking, small = whispering, show_to = speech_bubble_hearers, duration = 3 SECONDS)
+
+/atom/proc/send_runechat_to_atom_movable(message, speaking, small, list/show_to, duration = 3 SECONDS)
+	INVOKE_ASYNC(src, TYPE_PROC_REF(/atom/movable, animate_chat), message, speaking, small, show_to, duration)
```
Comment (Kapu1178) [inline code/game/atoms/atoms.dm:838]: y not just make `animate_chat()` waitfor = FALSE and avoid the 2 extra proc calls

File: code/modules/mob/floating_message.dm [hunk 2]
```diff
@@ -14,27 +19,45 @@ var/list/floating_chat_colors = list()
...
+		generate_floating_text(src, \
+		speaking.scramble(message), \
+		style, \
+		fontsize, \
+		duration, \
+		show_to, \
+		expression, \
+		colour = floating_chat_colors[src]
+	) \
+	: understood
```
Comment (Kapu1178) [inline code/modules/mob/floating_message.dm:60]: `\` isnt needed here.

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

**Comment by nevimer** [comment]
this branch has new commits that make it 👨‍🍳

File: code/modules/mob/floating_message.dm (hunk 2 above)
Comment (Zandario) [inline code/modules/mob/floating_message.dm:60]: Formatting Suggestion:
```suggestion
	var/image/understood = generate_floating_text(
		src,
		capitalize(message),
		style,
		fontsize,
		duration,
		show_to,
		expression,
		colour = floating_chat_colors[src],
	)
	var/image/gibberish = speaking ? generate_floating_text(
		src,
		speaking.scramble(message),
		style,
		fontsize,
		duration,
		show_to,
		expression,
		colour = floating_chat_colors[src],
	) : understood
```

File: code/modules/mob/floating_message.dm [hunk 3]
```diff
@@ -56,14 +79,24 @@ var/list/floating_chat_colors = list()
...
+	I.maptext = MAPTEXT_CENTER("<span style=\"[style]\">[message]</span>") // whoa calm down!!
+
 	for(var/image/old in holder.stored_chat_text)
 		animate(old, 2, pixel_y = old.pixel_y + 8)
 	LAZYADD(holder.stored_chat_text, I)
 
-	addtimer(CALLBACK(GLOBAL_PROC, .proc/remove_floating_text, holder, I), duration + 16)
-	addtimer(CALLBACK(GLOBAL_PROC, .proc/remove_images_from_clients, I, show_to), duration + 18)
+	addtimer(CALLBACK(GLOBAL_PROC, .proc/remove_floating_text, holder, I), duration + 24)
+	addtimer(CALLBACK(GLOBAL_PROC, .proc/remove_images_from_clients, I, show_to), duration + 28)
```
Comment (Zandario) [inline code/modules/mob/floating_message.dm:99]: 515 Compatibility Wrappers
```suggestion
	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(remove_floating_text), holder, I), duration + 24)
	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(remove_images_from_clients), I, show_to), duration + 28)
```

**Review by Zandario — COMMENTED** [review]
Just a couple things left imo

File: code/modules/mob/floating_message.dm (hunk 2 above)
Comment (nevimer) [reply code/modules/mob/floating_message.dm:60]: > Formatting Suggestion:
> 
> Reasoning for `if` over `?:`, if a ternary extends more than a line, readability suffers. And no there isn't a noticeable cost difference for `()?{}:{}` vs `if(){}else{}`
> ```suggestion
> 	var/image/understood = generate_floating_text(
> 		holder = src,
> 		message = capitalize(message),
> 		style = style,
> 		size = fontsize,
> 		duration = duration,
> 		show_to = show_to,
> 		expression = expression,
> 		colour = floating_chat_colors[src],
> 	)
> 	var/image/gibberish
> 	if(speaking)
> 		gibberish = generate_floating_text(
> 			holder = src,
> 			message = speaking.scramble(message),
> 			style = style,
> 			size = fontsize,
> 			duration = duration,
> 			show_to = show_to,
> 			expression = expression,
> 			colour = floating_chat_colors[src],
> 		)
> 	else
> 		gibberish = understood
> ```

The check is within the variable assignment, i don't think you can embed and if statement in those.

Comment (Lohikar) [reply code/modules/mob/floating_message.dm:60]: You can move the variable assignment into the if.

**Pull request closed without merging**